src/Profilers/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -247,28 +246,22 @@ HRESULT MainProfiler::InitializeCommandServer()
 
 HRESULT MainProfiler::MessageCallback(const IpcMessage& message)
 {
-    m_pLogger->Log(LogLevel::Debug, _LS("Message received from client %d"), message.Command);
+    m_pLogger->Log(LogLevel::Debug, _LS("Message received from client %hu:%hu"), message.CommandSet, message.Command);
 
-    switch (message.Command)
+    if (message.CommandSet == static_cast<unsigned short>(CommandSet::Profiler))
```

Comment:
Instead of special casing the profiler command set, could it be registered as a callback in `g_managedMessageCallbacks`? This would eliminate the specialization and possibly allow for consistent testing if unit testing were applied.

I would then suggest renaming `ManagedMessageCallbackManager` to `MessageCallbackManager` and `ManagedMessageCallback` to `MessageCallback`.

Reply:
I initially considered that, however doing so introducing a new set of considerations:

The `ManagedMessageCallbackManager` instance needs to be global so that PINVOKEs can access it. If we then place a callback in it for the profiler we'd effectively create a global reference to the `MainProfiler` instance. 

We could try to clear out the profiler's callback from `MessageCallbackManager` when we want to detach the profiler (or under some other stop conditions) but I'm personally not aware of all of the conditions when we'd want to do so. Having this special casing for the profiler command set was the most straight forward. Thoughts?

Reply:
> We could try to clear out the profiler's callback from MessageCallbackManager when we want to detach the profiler (or under some other stop conditions) but I'm personally not aware of all of the conditions when we'd want to do so.

We currently don't detach the profiler (although, we probably should, but that would a complete separate set of work that I would want us to wholistically think about), so I'm not too concerned about that.

There is a shutdown notion already where that can be plugged into: https://github.com/dotnet/dotnet-monitor/blob/main/src/Profilers/MonitorProfiler/MainProfiler/MainProfiler.cpp#L48

I'm okay with it as-is given that's how it was originally and there's nothing technically wrong with it. I'm usually an advocate for generalizing code when there's opportunity.
 
_Aside_
Ideally, the command server and handling would be completely separated from the profiler into its own dll/so such that any other dll/so can use it as a singleton, but that's well beyond the scope of this change. However, making this change (de-specialization of the command set) would move it slightly in that direction.